pyro/infer/predictive.py

```diff
@@ -63,7 +63,7 @@ def _predictive(model, posterior_samples, num_samples, return_sites=(),
         return trace
 
     return_site_shapes = {}
-    for site in model_trace.stochastic_nodes + model_trace.observation_nodes:
+    for site in model_trace.stochastic_nodes:
```

Comment:
@fehiepsi - To save travis runs for minor changes, I included this here. The motivation for this is that when we pass in observations to `Predictive` we shouldn't record `observation_nodes`. Does that seem reasonable to you (I forget the motivation for this, but it doesn't seem like we should need this now)?

Reply:
This is also exercised in `bayesian_regression_ii.ipynb`.

Reply:
This is at least useful for your deterministic `dist.Delta` trick. ;)

Reply:
Ahh..that's a good trick. Let us merge this if tests pass, and I'll address this in the other PR.

Reply:
okay, reverted.